File: Website/src/filters/ProductFilterContext.ts
import { FilterStrategy } from "./FilterStrategy";
import { Product } from "@/data/products";

export class ProductFilterContext {
  private strategy: FilterStrategy;

  constructor() {
    // default strategy that returns all products
    this.strategy = new (class implements FilterStrategy {
      execute(products: Product[]): Product[] {
        return products;
      }
    });
  }

  public setStrategy(strategy: FilterStrategy): void {
    this.strategy = strategy;
  }

  public executeFilter(products: Product[]): Product[] {
    return this.strategy.execute(products);
  }
}

File: Website/src/main/java/com/clofit/backend/MVC/ProductFilterContext.java
package com.clofit.backend.MVC;

import java.util.List;

import com.clofit.backend.model.Product;

public class ProductFilterContext {
    private IFilterStrategy strategy;

    public void setStrategy(IFilterStrategy strategy2) {
        this.strategy = strategy2;
    }

    public String getCurrentStrategyName() {
        if (strategy != null) {
            return strategy.getStrategyName();
        } else {
            return "No filter applied";
        }
    }

    public List<Product> executeFilter(List<Product> products) {
        if (strategy != null) {
            return strategy.filter(products);
        } else {
            return products;
        }
    }
}
